src/Core/View.js

```diff
@@ -458,4 +458,51 @@ View.prototype.execFrameRequesters = function execFrameRequesters(when, dt, upda
     }
 };
 
+const _eventCoords = new Vector2();
+/**
+ * Extract view coordinates from a mouse-event / touch-event
+ * @param {event} event - event can be a MouseEvent or a TouchEvent
+ * @param {number} touchIdx - finger index when using a TouchEvent (default: 0)
+ * @return {THREE.Vector2} - view coordinates (in pixels, 0-0 = top-left of the View)
+ */
+View.prototype.eventToViewCoords = function eventToViewCoords(event, touchIdx = 0) {
+    if (event.touches === undefined || !event.touches.length) {
+        return _eventCoords.set(event.offsetX, event.offsetY);
+    } else {
+        return _eventCoords.set(event.touches[touchIdx].clientX, event.touches[touchIdx].clientY);
```

Comment:
If I'm not mistaken that wouldn't work if the viewerDiv have an offset comparing to the viewport (aka in situation where the page is scrollable for example). We have viewport coordinates here, and we want client coordinates (relative to viewerDiv boundingRect).

The only possible solution I see is:
```js
const br = event.originalTarget.getClientBoundingRect(); // originalTarget should always be viewerDiv
return _eventCoords.set(event.touches[touchIdx].clientX - br.x, /* same for y */); 
```
I'm quite sure it does the right thing with margins and paddings (taking into account the first one, but not the second), but I'm not sure about borders. I still can't test this morning, but I'll have a touch device this afternoon.

Reply:
Or I can reuse the code from https://github.com/iTowns/itowns/commit/f3d77724200e651077ff0c9857d5e6b4e3d284f6#diff-df61a67eee71572718ef066564a3a0afR907 for touch events?

Reply:
Unless I miss something, I think my version is clearer and more correct? I don't understand why we use `offsetLeft` there, which is the offset to the parent element. We don't care about the parent element right? Beside I don't understand this 'static' test (and the name staticPos is actually reversed from its meaning).